### DISCUSSION TITLE
C++ code almost no highlighting

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [nvim-treesitter/nvim-treesitter#2096] [Q&A] [answered]
Should it be highlighted, i dont know....more?
It reminds me of regex highlighting. 
![Window_c++](https://user-images.githubusercontent.com/61395246/145607401-e4cd02d1-7647-411a-920e-fb5d692f54a2.png)

Is this normal?
`syntax is on`
`checkhealth is ok`
`cpp treesitter has all boxes ok`
Thanks!

**Comment by theHamsta** [comment, marked as answer]
It probably is regex highlighting. Can you try `:TSHighlightCapturesUnderCursor` from https://github.com/nvim-treesitter/playground. Also check `:TSModuleInfo`. Syntax will be usually switched off when tree-sitter highlighting activates.

**Reply by LamprosPitsillos** [reply]
TSModule output....
```
>> default       highlight  incremental_selection  indent                                                                                                                                      
                                                                                                                                       
cpp              ✗          ✓                      ✗  
```
CheckHealth output...
```
Parser/Features H L F I J                                                                                                                                                                   
                                                                                                                      
- cpp            ✓ ✓ ✓ ✓ ✓  
```
what is going on here?

**Reply by LamprosPitsillos** [reply]
`TSEnableEll highlight` enabled all of it.
`TSBufEnable highlight` fixed the buffer.
Have i been using TS all this time with highlighting off...?
```lua
-- TreeSitter
require("nvim-treesitter.configs").setup({
	incremental_selection = {
		enable = true,
	},
	ensure_installed = "maintained", -- one of "all", "maintained" (parsers with maintainers), or a list of languages
	ignore_install = { "javascript" }, -- List of parsers to ignore installing
	highlight = {}
})

```

**Reply by LamprosPitsillos** [reply]
ok..now i feel stupid...
I fix it just by adding `enable = true ` to highlight
```lua
...
highlight = {enable = true} 
...
```
thank you @theHamsta !
